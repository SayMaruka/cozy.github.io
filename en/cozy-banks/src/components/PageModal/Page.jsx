import React from 'react'
import cx from 'classnames'
import { Modal, ModalHeader, BarButton } from 'cozy-ui/transpiled/react'
import styles from 'components/PageModal/Page.styl'
import iconArrowLeft from 'assets/icons/icon-arrow-left.svg'

export const PageBackButton = ({ onClick }) => (
  <BarButton
    icon={iconArrowLeft}
    onClick={onClick}
    className={styles.Page__BackButton}
  />
)

export const PageHeader = ({ children }) => {
  return <ModalHeader className={styles.Page__header}>{children}</ModalHeader>
}

export const Page = ({ children, className, ...props }) => {
  return (
    <Modal
      mobileFullscreen
      closable={false}
      into="body"
      className={cx(styles.Page, className)}
      overlayClassName={styles.Page__Overlay}
      wrapperClassName={styles.Page__Wrapper}
      {...props}
    >
      {children}
    </Modal>
  )
}

export default Page
