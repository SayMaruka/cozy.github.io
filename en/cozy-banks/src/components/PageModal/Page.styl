@require 'settings/breakpoints'

BAR_HEIGHT=3rem
BAR_BUTTON_WIDTH=3rem
BAR_BORDER_WIDTH=1px

.Page__BackButton
    position absolute
    top 0
    left 0


.Page__header
    position relative
    height BAR_HEIGHT
    display flex
    align-items center
    justify-content center
    padding 0 BAR_BUTTON_WIDTH
    margin-bottom 0
    position relative
    font-weight bold

    font-size 1.25rem

    +small-screen()
        &
            font-size 1.125rem
            border-bottom 1px solid silver


.Page__Overlay,
.Page__Wrapper
    height 'calc(100% - -%s - %s)' % (BAR_HEIGHT BAR_BORDER_WIDTH)
