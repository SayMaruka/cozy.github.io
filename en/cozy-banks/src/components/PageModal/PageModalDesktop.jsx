import React from 'react'
import Modal, { ModalHeader, ModalContent } from 'cozy-ui/react/Modal'
import styles from 'components/PageModal/PageModalDesktop.styl'

const PageModalDesktop = props => {
  const { title, children, ...rest } = props

  return (
    <Modal
      {...rest}
      closeBtnClassName={styles.PageModalDesktop__cross}
      className={styles.PageModalDesktop}
    >
      <ModalHeader className={styles.PageModalDesktop__header}>
        {title}
      </ModalHeader>
      <ModalContent className={styles.PageModalDesktop__content}>
        {children}
      </ModalContent>
    </Modal>
  )
}

export default PageModalDesktop
