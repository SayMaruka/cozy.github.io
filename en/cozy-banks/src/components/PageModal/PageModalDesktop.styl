@require 'settings/breakpoints'

.PageModalDesktop
    .PageModalDesktop__header
        position relative
        border-bottom 1px solid var(--silver)
        margin-bottom 0
        padding-top 0
        padding-bottom 0
        height 5rem
        display flex
        align-items center
        justify-content center

.PageModalDesktop__content
    padding 0

    &:last-child
        padding 0

.PageModalDesktop__cross
    width 1rem
    height 1rem
    min-height auto
    z-index 1
    top 1.4rem
    right 1.4rem
