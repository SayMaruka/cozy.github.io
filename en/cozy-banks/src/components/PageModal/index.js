export { default as PageModal } from 'components/PageModal/PageModal'
