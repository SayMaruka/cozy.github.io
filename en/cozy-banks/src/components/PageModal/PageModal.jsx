import React from 'react'
import { withBreakpoints } from 'cozy-ui/transpiled/react'
import Page from 'components/PageModal/Page'
import { Modal } from 'cozy-ui/transpiled/react'

/**
 * This component renders a `Modal` on tablet and desktop viewports, and a
 * `Page` on mobile viewports. A `Page` is a special kind of fullscreen `Modal`
 * that shows the mobile bottom navigation bar when it's visible.
 *
 * `PageModal` takes the same props as
 * [cozy-ui's Modal](https://docs.cozy.io/cozy-ui/react/#modal).
 */
export const DumbPageModal = props => {
  const {
    breakpoints: { isMobile },
    ...rest
  } = props

  const Component = isMobile ? Page : Modal

  return <Component {...rest} />
}

export default withBreakpoints()(DumbPageModal)
