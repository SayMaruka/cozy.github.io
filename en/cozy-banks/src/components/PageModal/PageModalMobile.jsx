import React from 'react'
import Page from 'ducks/transactions/Page'

const PageModalMobile = props => {
  return <Page {...props} />
}

export default PageModalMobile
