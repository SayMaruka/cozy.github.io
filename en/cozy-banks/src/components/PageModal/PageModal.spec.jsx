import React from 'react'
import { shallow } from 'enzyme'
import { DumbPageModal } from './PageModal'

describe('PageModal', () => {
  it('should render a Modal on tablet/desktop', () => {
    const breakpoints = { isMobile: false }

    const el = shallow(
      <DumbPageModal breakpoints={breakpoints} title="Hello world">
        This is a page modal
      </DumbPageModal>
    ).getElement()

    expect(el).toMatchSnapshot()
  })

  it('should render a Page on mobile', () => {
    const breakpoints = { isMobile: true }

    const el = shallow(
      <DumbPageModal breakpoints={breakpoints} title="Hello world">
        This is a page modal
      </DumbPageModal>
    ).getElement()

    expect(el).toMatchSnapshot()
  })
})
