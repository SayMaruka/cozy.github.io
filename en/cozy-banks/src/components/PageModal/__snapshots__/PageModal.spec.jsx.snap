// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageModal should render a Modal on tablet/desktop 1`] = `
<Modal
  closable={true}
  into="body"
  mobileFullscreen={false}
  overflowHidden={false}
  primaryType="regular"
  secondaryType="secondary"
  size="small"
  title="Hello world"
>
  This is a page modal
</Modal>
`;

exports[`PageModal should render a Page on mobile 1`] = `
<Page
  title="Hello world"
>
  This is a page modal
</Page>
`;
